website/content/docs/agent-and-proxy/proxy/caching/static-secret-caching.mdx

```diff
@@ -0,0 +1,138 @@
+The auto-auth token must also have permissions to subscribe to KV event updates via Vault's [event system](/vault/docs/concepts/events).
+Proxy uses the auto-auth token to connect to Vault's event system, and uses events to know when to update its cache
+with updated secrets.
+
+An example policy snippet that would grant access to KVv1 and KVv2 events is given below:
+
+   ```hcl
+   path "sys/events/subscribe/kv*" {
+       capabilities = ["read"]
+   }
+   ```
+
+This functionality enables Proxy to receive updates as soon as a secret gets changed, reducing staleness in
+the cache, and means we only check for a secret update upon receiving notification that the secret got updated.
+
+Additionally, tokens accessing static secrets in the cache will need the ability to access [`sys/capabilies-self`](/vault/api-docs/system/capabilities-self)
+with the `update` capability. This capability allows a client to know what capabilities have on a current path, and allows
+a token to find out capabilities on multiple paths with a single request, instead of requiring the token to reach out to
+each explicitly and see if it gets a `403` response. This is used to efficiently maintain up-to-date access management
+on the cache. If a token has its access to a KV secret revoked, this will allow Proxy to know, so that it can prevent
+that token from accessing the cached secret.
+
+This permission is provided in the [`default` policy attached by default to all Vault tokens](/vault/docs/concepts/policies#default-policy),
+and, unless this has been modified, will be attached by default to all tokens. For completeness, or in case you wish to
+apply this permission in a way other than via the `default` policy, the policy snippet required for this functionality
+is as follows:
+
+   ```hcl
+    path "sys/capabilities-self" {
+        capabilities = ["update"]
+    }
+   ```
+
+The period at which capabilities for each token using Proxy will be checked is configured with the `static_secret_token_capability_refresh_interval`
+configuration, described below. The default is to check per token every five minutes. For a setup where the `api_proxy`
+has been configured with `use_auto_auth_token` as `true` or `force`, and all requests going through Proxy will be
+using this sole token, this will be a single request to Vault per this interval, no matter how many secrets are
+being cached.
+
+## Functionality
+
+When enabled, Vault Proxy will cache `GET`s to KVv1 and KVv2 endpoints. When a `GET` is made to a KV secret,
+we will cache the response in Proxy's cache. Once the response is cached, subsequent `GET`
+
+Vault Proxy uses the [event system](/vault/docs/concepts/events) to keep the cache up to date. It will
+pay attention to events that indicate both that the secret is one we have in our cache, and also that
+the secret has been modified (e.g. changed, deleted). Upon receiving such an event, it will update the cache
+to update or evict the cache entry as appropriate.
```

Comment:
🤔  This sounds more natural to me, but then again, I barely speak the language, haha. I am leaving the suggestion in case you prefer it too.
```suggestion
the secret has been modified (e.g. changed, deleted). Upon receiving such an event, it will update or evict the cache entry as appropriate.
```

Reply:
Thanks! I prefer your wording too. Updating!